src/withStyles.jsx

```diff
@@ -58,12 +58,9 @@ export function withStyles(
 
         const addedProps = {
           [themePropName]: ThemedStyleSheet.get(themeName),
+          [stylesPropName]: styleDef ? styleDef(themeName) : {},
```

Comment:
We should move the object up to a module-level variable so that it is always referentially equivalent between renders.